### ISSUE TITLE
[dev.icinga.com #13555] Move default configuration into separate package RPM

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#4865]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/13555

**Created by Boost on 2016-12-14 07:21:40 +00:00**

Assignee: _(none)_
Status: _New_ 
Target Version: _(none)_
Last Update: _2016-12-14 07:21:40 +00:00 (in Redmine)_

    Backport?: Not yet backported
    Include in Changelog: 1

---

I run icinga2 on a RPM based system (Fedora) and when upgrading to 2.6.0 my configuration does not work. Quick fix by running my Ansible playbook but it would still be nice if the configs would be left alone when upgrading.

My proposal is that all config is moved to package icinga2-configs and add its dependency to icinga2 package. Normal install would still get the same behaviour, but if I remove icinga2-config or exclude it from upgrade and roll my own an upgrade would be more likely to just work and not break anything.
The thing is I moved most of the files away from conf.d and into zones.d/global-templates and when rpm installs the upgrade, there are no files to consider overwriting so it writes the default configs to conf.d and breaks my setup.

It would also be nice to have a script to automate database upgrades for IDO.

**Comment by dnsmichi** [comment]
Blocked by #4799

**Comment by dnsmichi** [comment]
Note: On Windows, we shouldn't add anything in conf.d except for may the api-users.conf

**Comment by dnsmichi** [comment]
Should be superseded by #5605
